drivers/hmis/app/graphql/types/hmis_schema/query_type.rb

```diff
@@ -416,6 +416,26 @@ def form_definitions(filters:)
       scope.order(updated_at: :desc)
     end
 
+    field :form_identifier, Types::Forms::FormIdentifier, null: true do
+      argument :identifier, String, required: true
+    end
+    def form_identifier(identifier:)
+      raise 'Access denied' unless current_user.can_configure_data_collection?
+
+      Hmis::Form::Definition.where(identifier: identifier).current_versions.first
+    end
+
+    field :form_identifiers, Types::Forms::FormIdentifier.page_type, null: false do
+      filters_argument Forms::FormIdentifier
+    end
+    def form_identifiers(filters: nil)
+      raise 'Access denied' unless current_user.can_configure_data_collection?
+
+      scope = Hmis::Form::Definition.non_static.current_versions
+      scope = scope.apply_filters(filters) if filters
```

Comment:
@gigxz this filter which accepts `search_term` doesn't return hits if the search term matches previous versions. If an assessment used to be named 'XYZ' and its name changes to 'ABC', it won't be returned anymore if the user searches 'XYZ'.

To implement that I think we could
- make a new `Hmis::Filter::FormIdentifierFilter` (right now this is using the existing `Hmis::Filter::FormDefinitionFilter`)
- have it search all FormDefinitions for the search term
- then return the current version _for each of those ids_, a scope that doesn't exist currently

Do you agree we should implement this behavior and with this approach?